### PR TITLE
TMA-686: filtering segments in release brick

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/lcm/actions/segments_filter.rb [hunk 1]
```diff
@@ -29,16 +29,9 @@ def call(params)
...
-            data_product_segments_ids = data_product.segments.map(&:segment_id)
-
-            filtered_segments = filtered_segments.select do |segment|
-              data_product_segments_ids.include?(segment.segment_id.downcase)
-            end
-
             segments_filter = params.segments_filter.map(&:downcase)
 
-            filtered_segments = filtered_segments.select do |segment|
+            filtered_segments = params.segments.select do |segment|
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/segments_filter.rb]: Calls 'params.segments' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/segments_filter.rb]: Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/530/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/284/ : SUCCESS in 2m 13s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/591/ : SUCCESS in 2m 04s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/531/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1088/ : SUCCESS in 46m 29s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/847/ : SUCCESS in 23m 42s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/537/ : SUCCESS in 1m 21s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/289/ : SUCCESS in 1m 46s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/596/ : SUCCESS in 2m 03s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/196/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/298/ : SUCCESS in 27s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/605/ : SUCCESS in 29s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1097/ : SUCCESS in 44m 16s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/854/ : SUCCESS in 23m 00s

**Pull request merged**